test/integration_tests/python/test_jobs.py

```diff
@@ -94,3 +94,5 @@ def test_jobs(with_hierarchy, with_gear, with_invalid_gear, as_user, as_admin):
 
     r = as_user.post('/jobs/' + job1_id + '/retry')
     assert r.ok
+
+    db.jobs.remove()
```

Comment:
No, it's a very good point! I'll limit the scope of the `db.job.remove()`s I introduced. Sadly I can confirm that some assumptions like that are sprinkled here and there, though I think it's only a few.

I was thinking the DataBuilder could get some new features:
- store global db state (set of ids for all collections?) on init
- more flexibility for creation (enable users/gears/jobs + autocreate parent containers)
- keep track of resources created with the data_builder instance
- automatic deletion of resources (and any children thereof, even ones not created with this DataBuilder)
- assertions after deletion that the db state is unchanged

The global db state assertions could provide a unified way of making sure the tests aren't leaving anything behind and at the same time not deleting anything they didn't create. What do you think?